cnd/src/db/save.rs

```diff
@@ -516,3 +518,84 @@ impl Save<CreatedSwap<han::CreatedSwap, halight::CreatedSwap>> for Sqlite {
         unimplemented!()
     }
 }
+
+#[async_trait]
+impl Save<CreatedSwap<herc20::CreatedSwap, halight::CreatedSwap>> for Sqlite {
+    async fn save(
+        &self,
+        created: CreatedSwap<herc20::CreatedSwap, halight::CreatedSwap>,
+    ) -> anyhow::Result<()> {
+        let local_swap_id = created.swap_id;
+        let role = created.role;
+
+        // Save the swap into the swaps table.
+        let swap = ISwap::new(created.swap_id, created.peer, created.role);
+        self.save_swap(&swap).await?;
```

Comment:
I think all these things should happen within one DB transaction so they are atomic.